Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3062,7 +3061,7 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     auto applyDeclarativeNetRequestRules = [this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler), allJSONData = RetainPtr { allJSONData }] () mutable {
         if (!allJSONData.get().count) {
             removeDeclarativeNetRequestRules();
-            [declarativeNetRequestRuleStore() removeContentRuleListForIdentifier:uniqueIdentifier() completionHandler:^(NSError *) { }];
+            API::ContentRuleListStore::defaultStore().removeContentRuleListFile(declarativeNetRequestContentRuleListFilePath(), [](std::error_code) { });
```

Comment:
The completion handler should probably be captured in the lambda and only be called with true if the error code is empty.  Otherwise we're assuming that something that is dispatched to a different thread will always and synchronously immediately succeed.